src/saml.py

```diff
@@ -45,8 +48,23 @@ def __init__(self, charm_state: CharmState):
 
         Args:
             charm_state: The state of the charm that the Saml instance belongs to.
+
+        Raises:
+            CharmConfigInvalidError: if the certificate validation fails.
         """
         self._charm_state = charm_state
+        try:
+            if not self._is_certificate_valid:
```

Comment:
I'm not convinced this validation should be done in the class constructor, this is business logic, and business logic shouldn't be in __init__